src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -370,14 +397,20 @@ Terminal::UpdateSelectionParams Terminal::ConvertKeyEventToUpdateSelectionParams
 // - mods: the key modifiers pressed when performing this update
 void Terminal::UpdateSelection(SelectionDirection direction, SelectionExpansion mode, ControlKeyStates mods)
 {
+    // This is a special variable used to track if we should move the cursor when in mark mode.
+    //   We have special functionality where if you use the "switchSelectionEndpoint" action
+    //   when in mark mode (moving the cursor), we anchor an endpoint and you can use the
+    //   plain arrow keys to move the endpoint. This way, you don't have to hold shift anymore!
+    const bool markModeShouldMoveCursor = !_anchorSelectionEndpoint && !mods.IsShiftPressed();
+
     // 1. Figure out which endpoint to update
     // [Mark Mode]
     // - shift pressed --> only move "end" (or "start" if "pivot" == "end")
     // - otherwise --> move both "start" and "end" (moving cursor)
     // [Quick Edit]
     // - just move "end" (or "start" if "pivot" == "end")
     _selectionEndpoint = static_cast<SelectionEndpoint>(0);
-    if (_markMode && !mods.IsShiftPressed())
+    if (_markMode && markModeShouldMoveCursor)

```

Comment:
Does this behavior really depend on `_markMode`? To me it reads like `_anchorSelectionEndpoint` (and thus `markModeShouldMoveCursor`) is supposed to be sole dictator over whether we should engage this mode.

Reply:
Yeah, general keyboard selection does not let you move the cursor. That said, I only ever use this var as `_markMode && markModeShouldMoveCursor`. So I just threw it all into one variable and named it `shouldMoveCursor`.